anomalib/models/utils/blur.py

```diff
@@ -0,0 +1,76 @@
+"""Gaussian blurring via pytorch."""
+from typing import Tuple, Union
+
+from kornia.filters import get_gaussian_kernel2d
+from kornia.filters.filter import _compute_padding
+from kornia.filters.kernels import normalize_kernel2d
+from torch import Tensor, nn
+from torch.nn import functional as F
+
+
+class GaussianBlur2d(nn.Module):
```

Comment:
One might consider `GaussianBlur2d` as a post-processing step. Would it be an idea to move this `anomalib.post_processing`? @ORippler, @ashwinvaidya17, @djdameln ?

Reply:
Sure!

In my first understanding elements of `post_processing` were to be applied to the outputs of the complete `AnomalyModule`, which is why I didn't put the blur there initially (`AnomlyMapGenerators` were part of the `AnomalyModule` and thus the "main process" in my head, which is why blurring also wasn't a post-process).

Reply:
That's true. post_process is aimed to be applied to the output of the anomaly module. 

Gaussian blur, more specifically, is one of the final steps of the `AnomalyMapGenerator`, which is the final step of the anomaly module. That's why I thought that it could be considered as a post-processing step.

What do you guys think @ashwinvaidya17 @djdameln ?

Reply:
I agree that the `post_processing` should be reserved for operations that are applied to the output of the complete anomaly module. Also, the use-cases of GaussianBlur2d are not limited to the post-processing of the anomaly maps, but it could potentially be used in other parts of the model pipeline as well. That's why I would suggest to store it either in `anomalib/models/components` or `anomalib/utils`. I think my preference would go to the first option.

Reply:
I am also in favor of `anomalib/models/components`, maybe `anomlib/models/components/utils`

Reply:
Moving it to `anomalib/models/components` would make more sense, I agree. However, here is the tree structure of the components.

```bash
models/components
├── __init__.py
├── __pycache__
├── base
├── dimensionality_reduction
├── feature_extractors
├── freia
├── sampling
└── stats
```

So we cannot simply place it there. In this case, there would be need for either a new subpackage called such as `anomaly_map` to place anomaly map related components there. Or, as @ORippler suggested it could be `anomalib/models/components/utils`.

Reply:
How about `anomalib/models/components/filters`? If I am not wrong kornia stores gaussian blur under filters package. We can maybe follow their convention.

Reply:
I'm fine with `anomalib/models/components/filters`